### PR TITLE
Python data engineering challenge assignment for Wealth42

### PR DESCRIPTION
**Pull request by khannoaman**

### CODE REVIEW

File: Extract&Load.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    try:
+        response = requests.get(url)
+    except Exception as e:
+        print(e)
+        sys.exit()
+    data = response.json()
+
+    d = []
+    for bikepoint in data:
+        _id = bikepoint["id"][11:]
```
Comment (dwividha) [inline Extract&Load.py]: avoid hard coded constants like '11' without explanation - makes it difficult to understand the reason behind this

File: README.md [hunk 2]
```diff
@@ -1 +1,40 @@
...
+
+3. Now to create database "bikepoint" and then table "bikepoints data" inside that database run the file <strong><u>create_database.py</u></strong> by command
+
+	<pre>python create_database.py</pre>
+
+4. Next, to insert/update data into your database table "bikepoints data" run the file <strong><u>Extract&Load.py</u></strong> by command 
+	
+	<pre>python Extract&Load.py</pre>
+
+5. Now to update data every 5 minutes, we need to schedule the file <strong><u>Extract&Load.py</u></strong> to run every 5 minutes for that follow the steps below:
```
Comment (dwividha) [inline README.md:23]: Can you also include instructions using crontab?

File: Search_places.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+from sqlalchemy import create_engine
+from mysql_details import *
+import pandas as pd
+
+#Fetch Data from Database
+def get_data():
```
Comment (dwividha) [inline Search_places.py]: Fetching the whole table would be inefficient - can you instead craft a query which takes care of the common name matching and returns only the matching rows?

File: create_database.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        mycursor=conn.cursor()
+
+        #create Database
+        mycursor.execute("CREATE DATABASE bikepoints")
+
+        #use database bikepoints
+        mycursor.execute("USE bikepoints")
+
+        #create table "bikepoints data"  for storing real time bike availability data across London
+        mycursor.execute("""CREATE TABLE `bikepoints data` (
```
Comment (dwividha) [inline create_database.py]: why not use SQLAlchemy to create your table as well?

**Review by dwividha — CHANGES_REQUESTED** [review]

File: Extract&Load.py (hunk 1 above)
Comment (khannoaman) [reply Extract&Load.py]: Well, in the bikepoint api all the "id" keys have value like "BikePoints_1" so i am only storing the integer part which starts from 11th index. Thanks for your suggestion I'll keep it in my mind and explain it there as well.

File: create_database.py (hunk 4 above)
Comment (khannoaman) [reply create_database.py]: Actually, initially I started working with "mysql.connector" and later I realized that I need SQLALchemy engine object for storing table using pandas. I'll make changes and use SQLALchemy.

File: Search_places.py (hunk 3 above)
Comment (khannoaman) [reply Search_places.py]: Sure, I'll do that.

File: README.md (hunk 2 above)
Comment (khannoaman) [reply README.md:23]: okay, I'll add that too.

**Comment by khannoaman** [comment]
Changes required are done . 
Please re-review.

**Comment by dwividha** [comment]
Closing the PR

**Pull request closed without merging**